### ISSUE TITLE
Kelvin thinks that I’m changing the lights manually and it loops

### ISSUE THREAD

**Issue by handcoding** [stefanwichmann/kelvin#39]
I’m not sure if this might be related to issue #20, but Kelvin seems to think that I’m changing the lights manually even when I’m not, which results in Kelvin activating and disabling itself continuously. (I have the Hue scene set to “Reading (default Kelvin)” so that Kelvin will activate with the default schedule automatically.)

Here’s the output when I run Kelvin normally:

```
$ ./kelvin
INFO[2018/21/06 13:05:24] Kelvin v1.1.6 starting up... 🚀               
INFO[2018/21/06 13:05:24] Looking for updates...                       
INFO[2018/21/06 13:05:24] ⚙ Configuration config.json loaded           
INFO[2018/21/06 13:05:24] ⌘ Connection to bridge established           
INFO[2018/21/06 13:05:24] ⌘ Devices found on current bridge:           
INFO[2018/21/06 13:05:24] | Name                 |  ID | On    | Dimmable | Temperature | Color | 
INFO[2018/21/06 13:05:24] | Small desk lamp      |   1 | true  | true     | true        | true  | 
INFO[2018/21/06 13:05:24] | Large desk lamp      |   2 | true  | true     | true        | true  | 
INFO[2018/21/06 13:05:24] | Patio lamp           |   3 | true  | true     | true        | true  | 
INFO[2018/21/06 13:05:24] | Bedside lamp         |   5 | true  | true     | true        | true  | 
INFO[2018/21/06 13:05:24] 🌍 Working with location 32.9143, -96.7366 from configuration 
INFO[2018/21/06 13:05:25] 💡 Light Large desk lamp - Activating schedule for Jun 21 2018 (Sunrise: 06:18, Sunset: 20:38) 
INFO[2018/21/06 13:05:25] 💡 Light Bedside lamp - Activating schedule for Jun 21 2018 (Sunrise: 06:18, Sunset: 20:38) 
INFO[2018/21/06 13:05:25] 💡 Light Bedside lamp - Activating interval 06:18 - 20:38 
INFO[2018/21/06 13:05:25] 💡 Light Patio lamp - Activating schedule for Jun 21 2018 (Sunrise: 06:18, Sunset: 20:38) 
INFO[2018/21/06 13:05:25] 💡 Light Patio lamp - Activating interval 06:18 - 20:38 
INFO[2018/21/06 13:05:25] 💡 Light Large desk lamp - Activating interval 06:18 - 20:38 
INFO[2018/21/06 13:05:25] 💡 Light Small desk lamp - Activating schedule for Jun 21 2018 (Sunrise: 06:18, Sunset: 20:38) 
INFO[2018/21/06 13:05:25] 💡 Light Small desk lamp - Activating interval 06:18 - 20:38 
INFO[2018/21/06 13:05:26] 💡 Light Patio lamp - Light just appeared.    
INFO[2018/21/06 13:05:26] 💡 Light Patio lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:26] 💡 Light Small desk lamp - Light just appeared. 
INFO[2018/21/06 13:05:26] 💡 Light Small desk lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:26] 💡 Light Large desk lamp - Light just appeared. 
INFO[2018/21/06 13:05:26] 💡 Light Large desk lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:26] 💡 Light Bedside lamp - Light just appeared.  
INFO[2018/21/06 13:05:26] 💡 Light Bedside lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:27] 💡 Light Bedside lamp - Light state has been changed manually. Disabling Kelvin... 
INFO[2018/21/06 13:05:27] 💡 Light Large desk lamp - Light state has been changed manually. Disabling Kelvin... 
INFO[2018/21/06 13:05:27] 💡 Light Small desk lamp - Light state has been changed manually. Disabling Kelvin... 
INFO[2018/21/06 13:05:27] 💡 Light Patio lamp - Light state has been changed manually. Disabling Kelvin... 
INFO[2018/21/06 13:05:28] 💡 Light Large desk lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:28] 💡 Light Small desk lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:28] 💡 Light Bedside lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:28] 💡 Light Patio lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:29] 💡 Light Bedside lamp - Light state has been changed manually. Disabling Kelvin... 
INFO[2018/21/06 13:05:29] 💡 Light Large desk lamp - Light state has been changed manually. Disabling Kelvin... 
INFO[2018/21/06 13:05:29] 💡 Light Patio lamp - Light state has been changed manually. Disabling Kelvin... 
INFO[2018/21/06 13:05:29] 💡 Light Small desk lamp - Light state has been changed manually. Disabling Kelvin... 
INFO[2018/21/06 13:05:30] 💡 Light Patio lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:30] 💡 Light Large desk lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:30] 💡 Light Small desk lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:30] 💡 Light Bedside lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:31] 💡 Light Large desk lamp - Light state has been changed manually. Disabling Kelvin... 
INFO[2018/21/06 13:05:31] 💡 Light Small desk lamp - Light state has been changed manually. Disabling Kelvin... 
INFO[2018/21/06 13:05:31] 💡 Light Patio lamp - Light state has been changed manually. Disabling Kelvin... 
INFO[2018/21/06 13:05:31] 💡 Light Bedside lamp - Light state has been changed manually. Disabling Kelvin... 
INFO[2018/21/06 13:05:32] 💡 Light Patio lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:32] 💡 Light Small desk lamp - Detected matching target state. Activating Kelvin... 
INFO[2018/21/06 13:05:32] 💡 Light Bedside lamp - Detected matching target state. Activating Kelvin... 

```

And if it may help, here’s the output from Kelvin when I run it with `./kelvin --debug`:
[Kelvin - debug log.txt](https://github.com/stefanwichmann/kelvin/files/2124828/Kelvin.-.debug.log.txt)

P.S. I’m using a combination of 2nd generation Hue White and Color Ambiance bulbs and 3rd generation Hue White and Color Ambiance bulbs. And I’ve also double-checked that they have the latest firmware.

**Comment by stefanwichmann** [comment]
Hi Ashley,

thanks for this really detailed and complete bug report! I had a look at your debug output and think I know what's going on.

I am currently trying to reproduce your scenario to see if my fix is working. Stay tuned...

**Comment by stefanwichmann** [comment]
Hi Ashley,

sadly I could not reproduce the bug you are experiencing. So my fix is kind of a shot into the blue. I released it as part of Kelvin 1.1.8 anyway...

Please give it a try and let me know if it works!

**Comment by handcoding** [comment]
@stefanwichmann You betcha—I’ll give it a shot and I’ll report back!
